src/Apps/Legion/Components/WizardShell/Internal/DataSourceStep/DataSourceStep.tsx

```diff
@@ -18,26 +65,342 @@ const getClassNames = classNamesFunction<
+            getTablesState.callAdapter({
+                databaseName: option.text
+            });
+        },
+        [getTablesState]
+    );
+    const handleTargetDatabaseChange = useCallback(
+        (newValue: any, actionMeta: ActionMeta<any>) => {
+            dispatch({
+                type: DataSourceStepActionType.SET_SELECTED_TARGET_DATABASE,
+                database: newValue
+            });
+
+            if (actionMeta.action === 'create-option') {
+                setAdapterResult(null);
+                createDatabaseState.callAdapter({
+                    databaseName: newValue.label
+                });
+                dispatch({
+                    type: DataSourceStepActionType.SET_TARGET_DATABASE_OPTIONS,
+                    options: state.targetDatabaseOptions.concat(newValue)
+                });
+            }
+        },
+        [createDatabaseState, state.targetDatabaseOptions]
+    );
+    const handleSourceTableChange = useCallback(
+        (_event, option: IDropdownOption) => {
+            dispatch({
+                type: DataSourceStepActionType.SET_SELECTED_SOURCE_TABLE,
+                table: option.text
+            });
+            getTableState.callAdapter({
+                databaseName: state.selectedSourceDatabase,
+                tableName: option.text
+            });
+        },
+        [getTableState, state.selectedSourceDatabase]
+    );
+    const handleSourceTwinIDColumnChange = useCallback(
+        (_event, option: IDropdownOption) => {
+            dispatch({
+                type:
+                    DataSourceStepActionType.SET_SELECTED_SOURCE_TWIN_ID_COLUMN,
+                columnName: option.text
+            });
+        },
+        []
+    );
+    const handleSourceTableTypeChange = useCallback(
+        (_event, option: IDropdownOption) => {
+            dispatch({
+                type: DataSourceStepActionType.SET_SELECTED_SOURCE_TABLE_TYPE,
+                tableType: option.key as string
+            });
+        },
+        []
+    );
+    const handleCookButtonClick = useCallback(() => {
+        setCookAssets(
+            cookSourceTable(
+                `${adapter.connectionString}/${state.selectedSourceDatabase}/${state.selectedSourceTable}`,
+                state.sourceTableData,
+                state.selectedSourceTwinIDColumn,
+                state.selectedSourceTableType as TableTypes
+            )
+        );
+    }, [
+        adapter.connectionString,
+        state.selectedSourceDatabase,
+        state.selectedSourceTable,
+        state.sourceTableData,
+        state.selectedSourceTwinIDColumn,
+        state.selectedSourceTableType
+    ]);
+
+    const handleNextClick = () => {
+        const viewModels = getViewModelsFromCookedAssets(cookAssets.models);
+        const viewTwins = getViewTwinsFromCookedAssets(
+            cookAssets.twins,
+            viewModels
+        );
+        wizardNavigationContextDispatch({
```

Comment:
TODO (for me) merge these 3 dispatch functions into a single one called `NEXT_STEP`. Allow NEXT_STEP to update data and navigate to the next step.